src/api/docgen/view/api-reference.ts

```diff
@@ -0,0 +1,58 @@
+import * as reflect from "jsii-reflect";
+import { Markdown } from "../render/markdown";
+import { Transpile } from "../transpile/transpile";
+import { Classes } from "./classes";
+import { Constructs } from "./constructs";
+import { Enums } from "./enums";
+import { Interfaces } from "./interfaces";
+import { Structs } from "./structs";
+
+/**
+ * Render an API reference based on the jsii assembly.
+ */
+export class ApiReference {
+  private readonly constructs: Constructs;
+  private readonly structs: Structs;
+  private readonly interfaces: Interfaces;
+  private readonly classes: Classes;
+  private readonly enums: Enums;
+  constructor(
+    transpile: Transpile,
+    assembly: reflect.Assembly,
+    submodule?: reflect.Submodule
+  ) {
+    const classes = this.filterSubmodule(assembly.classes, submodule);
+    const interfaces = this.filterSubmodule(assembly.interfaces, submodule);
+    const enums = this.filterSubmodule(assembly.enums, submodule);
+
+    this.constructs = new Constructs(transpile, classes);
+    this.classes = new Classes(transpile, classes);
+    this.structs = new Structs(transpile, interfaces);
+    this.interfaces = new Interfaces(transpile, interfaces);
+    this.enums = new Enums(transpile, enums);
+  }
+
+  /**
+   * Generate markdown.
+   */
+  public get markdown(): Markdown {
```

Comment:
Personally I prefer not to use a getter API for something that preforms processing. I'd call this `render(): Markdown`

Reply:
changed to `render`